bin/migrate_to_single_table.py

```diff
@@ -0,0 +1,319 @@
+            self.logger.info(f"Number of records in new table: {new_total_records}")
+            if total_records == new_total_records:
+                self.logger.info(
+                    "Number of records in the new table matches the number of records in old table"
+                )
+            else:
+                self.logger.info(
+                    "Number of records in the new table DOES NOT MATCH the number of records in old table. Check logs to see if there are records that were not migrated"
+                )
+            self.logger.info("Finished migrating :D")
+
+    def get_index_record_hash(self, did):
+        """
+        Get the index record hash for the given did and return correctly formatted value
+        """
+        try:
+            stmt = (
+                self.session.query(
+                    IndexRecordHash.hash_type,
+                    IndexRecordHash.hash_value,
+                )
+                .filter(IndexRecordHash.did == did)
+                .all()
+            )
+            res = {hash_type: hash_value for hash_type, hash_value in stmt}
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with hash for {did}: {e}")
+
+    def get_urls_record(self, did):
+        """
+        Get the urls record for the given did and return correctly formatted value
+        """
+        try:
+            stmt = (
+                self.session.query(IndexRecordUrl.url)
+                .filter(IndexRecordUrl.did == did)
+                .all()
+            )
+            res = [u.url for u in stmt]
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with urls for {did}: {e}")
+
+    def get_urls_metadata(self, did):
+        """
+        Get the urls metadata for the given did and return correctly formatted value
+        """
+        try:
+            stmt = (
+                self.session.query(
+                    IndexRecordUrlMetadata.url,
+                    IndexRecordUrlMetadata.key,
+                    IndexRecordUrlMetadata.value,
+                )
+                .filter(IndexRecordUrlMetadata.did == did)
+                .all()
+            )
+            res = {url: {key: value} for url, key, value in stmt}
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with url metadata for {did}: {e}")
+
+    def get_index_record_ace(self, did):
+        """
+        Get the index record ace for the given did and return correctly formatted value
+        """
+        try:
+            stmt = (
+                self.session.query(IndexRecordACE.ace)
+                .filter(IndexRecordACE.did == did)
+                .all()
+            )
+            res = [a.ace for a in stmt]
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with ace for did {did}: {e}")
```

Comment:
are you intending to not reraise an error here? so we just skip that field and it gets set as None? I don't think we want that behavior b/c then some records could get "successfully" submitted but if one of these functions failed, that field would be None. This is a good reason to try and avoid early returns b/c that behavior is not obvious. But if you had a guaranteed return value at the end, it would be.

e.g.

```
field_value = None
try:
  get
except: 
  whoops
return field_value
```

That being said, I don't think we want that behavior. I think we want to raise the exception higher and then exponentially backoff trying 1 or 2 more times to get the info for the record.

Reply:
Yeah but these aren't catching the exception just because the fields are [], right? Like urls can be empty, but `get_urls_record` should still return []. I'm talking about the case where there really is an exception and you catch it.

The db query fails b/c we lose connection to the db. So right now you log that, but you don't return anything or reraise the exception, so the code continues on. I don't _think_ we want that behavior. If the db connection is broken, we want to raise an exception. And higher up, if we get an exception, we want to retry getting the record a few times just in case it was something flaky.